common/djangoapps/util/enterprise_api.py

```diff
@@ -0,0 +1,62 @@
+"""
+Utilities to communicate with the Enterprise service.
+"""
+import logging
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from edx_rest_api_client.client import EdxRestApiClient
+from slumber.exceptions import HttpClientError, HttpServerError
+
+from openedx.core.lib.token_utils import JwtBuilder
+
+
+log = logging.getLogger(__name__)
+
+
+class EnterpriseApiException(Exception):
+    """
+    Exception for errors while communicating with the Enterprise service API.
+    """
+    pass
+
+
+class EnterpriseApiClient(object):
+    """
+    Class for producing an Enterprise service API client.
+    """
+
+    def __init__(self):
+        """
+        Initialize an Enterprise service API client, authenticated using the Enterprise worker username.
+        """
+        self.user = User.objects.get(username=settings.ENTERPRISE_SERVICE_WORKER_USERNAME)
+        jwt = JwtBuilder(self.user).build_token([])
+        self.client = EdxRestApiClient(
+            settings.ENTERPRISE_API_URL,
+            jwt=jwt
+        )
+
+    def post_enterprise_course_enrollment(self, username, course_id, consent_granted):
+        """
+        Create an EnterpriseCourseEnrollment by using the corresponding serializer (for validation).
+        """
+        data = {
+            'username': username,
+            'course_id': course_id,
+            'consent_granted': consent_granted,
+        }
+        endpoint = getattr(self.client, 'enterprise-course-enrollment')  # pylint: disable=literal-used-as-attribute
```

Comment:
We can, although I think that goes against the spirit of what that violation is actually trying to discourage. It's encouraging the use of the dot operator over `getattr` wherever possible. By us disabling it here, we're explicitly saying that the violation doesn't make sense in this case - we have no choice but to use `getattr` because the attribute name is not a valid python name (contains `-`).